File: Backend/Models/Emailconfiguration.cs
using System;

namespace SkyVault.Models;

public partial class Emailconfiguration
{
    public Guid Emailconfigurationid { get; set; }

    public string Smtphost { get; set; } = null!;

    public int Smtpport { get; set; }

    public bool Usessl { get; set; }

    public bool Requiresauthentication { get; set; }

    public string Senderemail { get; set; } = null!;

    public string? Senderdisplayname { get; set; }

    public string? Username { get; set; }

    public string? Encryptedpassword { get; set; }

    public bool Isactive { get; set; }

    public DateTime Createdat { get; set; }

    public DateTime Updatedat { get; set; }
}

File: Frontend/src/models/emailConfiguration/EmailConfiguration.ts
export interface CreateEmailConfigurationRequest {
  smtpHost: string
  smtpPort: number
  useSsl: boolean
  requiresAuthentication: boolean
  senderEmail: string
  senderDisplayName: string | null
  username: string | null
  password: string | null
  isActive: boolean
}

export interface UpdateEmailConfigurationRequest {
  smtpHost: string
  smtpPort: number
  useSsl: boolean
  requiresAuthentication: boolean
  senderEmail: string
  senderDisplayName: string | null
  username: string | null
  password: string | null
}

export interface EmailConfigurationResponse {
  emailConfigurationId: string
  smtpHost: string
  smtpPort: number
  useSsl: boolean
  requiresAuthentication: boolean
  senderEmail: string
  senderDisplayName: string | null
  username: string | null
  isActive: boolean
  createdAt: string
  updatedAt: string
}
